examples/tokio02_async_smtp.rs

```diff
@@ -0,0 +1,32 @@
+// This line is only to make it compile from lettre's examples folder,
+// since it uses Rust 2018 crate renaming to import tokio.
+// Won't be needed in user's code.
+use tokio02_crate as tokio;
+
+use lettre::transport::smtp::authentication::Credentials;
+use lettre::{AsyncSmtpTransport, Message, Tokio02Connector, Tokio02Transport};
+
+#[tokio::main]
+async fn main() {
+    let email = Message::builder()
+        .from("NoBody <nobody@domain.tld>".parse().unwrap())
+        .reply_to("Yuin <yuin@domain.tld>".parse().unwrap())
+        .to("Hei <hei@domain.tld>".parse().unwrap())
+        .subject("Happy new async year")
+        .body("Be happy with async!")
+        .unwrap();
+
+    let creds = Credentials::new("smtp_username".to_string(), "smtp_password".to_string());
+
+    // Open a remote connection to gmail
+    let mailer = AsyncSmtpTransport::<Tokio02Connector>::relay("smtp.gmail.com")
```

Comment:
I think I found a way of seamlessly supporting multiple runtimes. We'll probably have to document it, because omitting the `::<Tokio02Connector>` part doesn't give a straight forward suggestion, but the good thing is that it will be possible to simply replace `Tokio02Connector` with `Tokio03Connector` or `AsyncStd1Connector` (once we add support for them) and that will be it. No other code to change other than changing imports and updating features, and users will have switched runtime